src/rational/q/from.rs

```diff
@@ -249,3 +343,105 @@ mod tests_from_str {
         assert_eq!(zero_1, zero_2);
     }
 }
+
+#[cfg(test)]
+mod test_from_int_int {
+
+    use crate::integer::Z;
+    use crate::rational::Q;
+
+    /// Test the different borrowed parameter types
+    #[test]
+    fn test_types_borrowed() {
+        let numerator = 10;
+        let denominator = 15;
+
+        let q_1 = Q::try_from((&(numerator as u8), &(denominator as i8))).unwrap();
+        let q_2 = Q::try_from((&(numerator as u16), &(denominator as i16))).unwrap();
+        let q_3 = Q::try_from((&(numerator as u32), &(denominator as i32))).unwrap();
+        let q_4 = Q::try_from((&(numerator as u64), &(denominator as i64))).unwrap();
+        let q_5 = Q::try_from((&Z::from(numerator), &Z::from(denominator))).unwrap();
```

Comment:
Please also add tests where the numerator is a signed integer and the denominator is unsigned. It does not have to match all combinations, but at least all combinations (sigend, unsigned), (unsigned, unsigned), (signed, signed) should be added with one example

Reply:
Added